scenarios/kubernetes_e2e.py

```diff
@@ -52,7 +55,58 @@ def kubekins(tag):
     """Return full path to kubekins-e2e:tag."""
     return 'gcr.io/k8s-testimages/kubekins-e2e:%s' % tag
 
+# pylint: disable=too-few-public-methods
+class BoskosProject(object):
+    """Loan/Return a gcp project from boskos service."""
+    def __init__(self, job, timeout):
+        self._job = job
+        self._timeout = timeout
+        self._project = None
 
+    def __enter__(self):
+        self._project = self._request(self._job, self._timeout)
+        return self._project
+
+    def __exit__(self, *args):
+        if self._project and self._job:
+            self._return(self._job)
+
+    @property
+    def project(self):
+        """Getter for project name."""
+        return self._project
+
+    @staticmethod
+    def _request(job, timeout):
+        """Load a project from boskos."""
+        if not job or not timeout:
+            return None
+        url = 'http://boskos/request?job=%s&duration=%s' % (job, timeout)
+        for attempt in range(0, 3):
+            res = urllib2.urlopen(url)
+            if res.getcode() == 200:
+                data = json.loads(res)
+                return data['Name']
+            elif res.getcode() == 204: # StatusNoContent, no available projects
+                sleep(60) # Wait a min and try our luck later.
+            else:
+                print >>sys.stderr, 'Request return with %d' % res.getcode()
+                raise ValueError(res.getcode(), 'Failed request project!')
+            print >>sys.stderr, 'Will retry %d times' % (3-attempt)
+        return None
+
+    @staticmethod
+    def _return(job):
+        """Return the loaner back to boskos."""
+        url = "http://boskos/return?job=%s" % job
```

Comment:
How many minutes will jobs reserve this project?

If the VM running the job blows up 1m after reserving the project, how many minutes does it take the system to detect and recover from this situation?